operators/src/source/gdal_source/mod.rs

```diff
@@ -467,116 +426,68 @@ where
-                &rasterband,
-                &dataset_grid_bounds,
-                tile_grid_bounds,
-                no_data_value,
-            )?;
-
-            let mut tile_raster = Grid2D::new_filled(output_shape, fill_value, no_data_value);
-            tile_raster.grid_blit_from(dataset_raster);
-            tile_raster.into()
-        };
+        let result_tile = read_raster_tile_with_properties(
+            &dataset,
+            dataset_params,
+            tile_information,
+            tile_time,
+        )?
+        .unwrap_or_else(|| create_no_data_tile(tile_information, tile_time, no_data_value));
 
         let elapsed = start.elapsed();
-        debug!("data loaded -> returning data grid, took {:?}", elapsed);
+        tracing::debug!("data loaded -> returning data grid, took {:?}", elapsed);
 
-        Ok(GridWithProperties {
-            grid: result_grid,
-            properties,
-        })
+        Ok(result_tile)
     }
 
     ///
-    /// A stream of `RasterTile2D`
+    /// A stream of futures producing `RasterTile2D` for a single slice in time
     ///
-    pub fn tile_stream(
-        &self,
+    fn temporal_slice_tile_future_stream<T: Pixel + GdalType>(
         query: RasterQueryRectangle,
-        info: GdalLoadingInfoPart,
-    ) -> impl Stream<Item = Result<RasterTile2D<T>>> {
-        let spatial_resolution = query.spatial_resolution;
-        let no_data_value = self.no_data_value;
-
-        // TODO: evaluate if there are GeoTransforms with positive y-axis
-        // The "Pixel-space" starts at the top-left corner of a `GeoTransform`.
-        // Therefore, the pixel size on the x-axis is always increasing
-        let pixel_size_x = spatial_resolution.x;
-        debug_assert!(pixel_size_x.is_sign_positive());
-        // and the pixel size on  the y-axis is always decreasing
-        let pixel_size_y = spatial_resolution.y * -1.0;
-        debug_assert!(pixel_size_y.is_sign_negative());
-
-        let tiling_strategy = self
-            .tiling_specification
-            .strategy(pixel_size_x, pixel_size_y);
-
-        let source_stream =
-            stream::iter(tiling_strategy.tile_information_iterator(query.spatial_bounds))
-                .map(move |tile| {
-                    Self::load_tile_async(info.params.clone(), tile, info.time, no_data_value)
-                })
-                .buffered(1); // TODO: find a good default and / or add to config.
-
-        // use SparseTilesFillAdapter to fill all the gaps
-        SparseTilesFillAdapter::new(
-            source_stream,
-            tiling_strategy.tile_grid_box(query.spatial_partition()),
-            tiling_strategy.geo_transform,
-            tiling_strategy.tile_size_in_pixels,
-            no_data_value.unwrap_or_else(T::zero),
+        info: GdalLoadingInfoTemporalSlice,
+        no_data_value: Option<T>,
+        tiling_strategy: TilingStrategy,
+    ) -> impl Stream<Item = impl Future<Output = Result<RasterTile2D<T>>>> {
```

Comment:
Why `Stream` of `Future`s? You can flatten it to have just a stream of the data, right?

Reply:
we need the futures because this is how buffered works: https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.buffered